breacharbiter.go

```diff
@@ -907,61 +989,101 @@ func (b *breachArbiter) createJusticeTx(
 func (b *breachArbiter) craftCommitSweepTx(
 	closeInfo *lnwallet.UnilateralCloseSummary) (*wire.MsgTx, error) {
 
-	// First, we'll fetch a fresh script that we can use to sweep the funds
-	// under the control of the wallet.
-	sweepPkScript, err := newSweepPkScript(b.wallet)
+	selfOutput := newBreachedOutput(
+		closeInfo.SelfOutPoint,
+		lnwallet.CommitmentNoDelay,
+		closeInfo.SelfOutputSignDesc,
+	)
+
+	// Begin with a base txn weight of 4 * tx_non_wit_data +
+	// witness_header_size.
+	var txWeight uint64 = 4*53 + 2
+	// Add receiver script witness and tx input
+	txWeight += 325 + 4*41
+
+	return b.sweepSpendableOutputsTxn(txWeight, selfOutput)
+}
+
+// sweepSpendableOutputsTxn creates a signed transaction from a sequence of
+// spendable outputs by sweeping the funds into a single p2wkh output.
+func (b *breachArbiter) sweepSpendableOutputsTxn(
+	txWeight uint64,
```

Comment:
It seems that this function accounts for the weight of the inputs added (with their witnesses), but _not_ the weight of the added outputs.

Reply:
To remedy this, we can instead modify the param to be the `totalWitnessWeight` alone. Then within this function, we can craft the sweep transaction and compute the total weight of that. The final weight we'll use for fee estimation is the sum of both of these computed weights.